benchmarks/benchmark_gpucsl.py

```diff
@@ -115,16 +111,14 @@ def run_benchmark_gaussian(samples, max_level, devices, sync_device):
         max_level, samples.shape[1], devices
     )
 
-    (pc_result, pc_runtime) = pc(
+    pc = GaussianPC(
         samples,
-        DataDistribution.GAUSSIAN,
         max_level,
         0.05,
-        gaussian_correlation_matrix=correlation_matrix,
         kernels=kernels,
-        devices=devices,
-        sync_device=sync_device,
-    )
+    ).set_distribution_specific_options(devices, sync_device, correlation_matrix)
+
+    (pc_result, pc_runtime) = pc.execute()
```

Comment:
Would it be possible, to call `set_distribution_specific_options` in the constructor of `GaussianPC`, rather than calling it separately?

Reply:
not really, as we do not know which of the given parameters are needed (we would pass more than intended).
Here two variants I could come up with:
1. overwrite the __init__ method in every subclass, something I find to be annoying at this amount of paramters already.
2. I could also return the set_distribution_specific_options method from the constructor, so you could call: `GaussianPC(samples, ...)(devices, ...)`

Maybe you like one of this two methods (if we change something I would prefer

Reply:
Ok, then maybe it makes more sense to leave it as it is